lib/linguist/heuristics.yml

```diff
@@ -28,6 +28,29 @@
 #
 ---
 disambiguations:
+- extensions: ['.1', '.1in', '.1m', '.1x', '.2', '.3', '.3in', '.3m', '.3qt', '.3x', '.4', '.5', '.6', '.7', '.8', '.9', '.man', '.mdoc']
+  rules:
+  # ".so links" are used like symlinks in manpage directories.
+  # Match only if target file uses a relevant file extension.
+  - language: Roff Manpage
+    pattern: '\A[.''][ \t]*so[ \t]+\S+?\.((?:[1-9](?![0-9])[a-z_0-9]*|0p|n|man|mdoc)(?:\.in)?)\s*\z'
```

Comment:
Wouldn't it be better (and simpler and safer) to show the typical roff files containing
    nothing but a single .so line verbatim rather than trying to render the file pointed to?
    If the user wants to see the rendered manual page, they can always look at the target file.
    It seems valuable to clearly show the file containing .so as a roff-style symlink.

    Besides, i see nothing here requiring that the file only contains the single line.
    So any file containing .so pointing to a man-page-style filename will be treated
    as a manual page, even if there is much other stuff in the file before or after the .so?

    Finally, calling mandoc(1) on such a link in the present context is very likely to fail.
    These links typically read ".so man1/foo.1" but there is almost certainly no "man1"
    directory in the source tree, so mandoc(1) cannot include the target file.  Besides,
    mandoc is very picky, for security reasons, about which .so paths it will follow, and
    if the path looks odd, it will render something like  "See the file man1/foo.1" which
    is less useful than seeing the raw file content.

Reply:
> If the user wants to see the rendered manual page, they can always look at the target file.

Okay, yeah, that's true. 😀 And the part about failing based on include paths is a good point, too. Will remove. 👍

> Besides, i see nothing here requiring that the file only contains the single line.

The `\A` and `\z` anchors are like `^` and `$`, except they're guaranteed to match only the start and end of the *entire* input, respectively.